xapian-core/cluster/clusterinternal.h

```diff
@@ -87,16 +85,19 @@ class Xapian::Cluster::Internal : public Xapian::Internal::intrusive_base {
     Centroid centroid;
 
   public:
+    /// Constructor that initialises cluster with centroid
     Internal(const Centroid &centroid_) : centroid(centroid_) {}
```

Comment:
I missed this before, but a constructor which can take a single argument should be marked `explicit` (unless you want the compiler to be able to implicitly convert a `Centroid` to a `Cluster::Internal` by invoking this constructor).